tests/unit/adapters/application-test.js

```diff
@@ -29,103 +29,187 @@ moduleFor('adapter:application', 'Unit | Adapter | application', {
 });
 
 test('#find calls #findOne when options arg is a string', function(assert) {
+  assert.expect(3);
+  const done = assert.async();
+
   const adapter = this.subject();
   sandbox.stub(adapter, 'findOne', function () { return RSVP.Promise.resolve(null); });
   let promise = adapter.find('1');
+
   assert.ok(typeof promise.then === 'function', 'returns a thenable');
-  assert.ok(adapter.findOne.calledOnce, 'findOne called');
-  assert.ok(adapter.findOne.calledWith('1'), 'findOne called with "1"');
+  promise.then(() => {
+    assert.ok(adapter.findOne.calledOnce, 'findOne called');
+    assert.ok(adapter.findOne.calledWith('1'), 'findOne called with "1"');
+    done();
+  });
 });
 
 test('#find calls #findOne when options arg is an object having an id property', function(assert) {
+  assert.expect(3);
+  const done = assert.async();
+
   const adapter = this.subject();
   sandbox.stub(adapter, 'findOne', function () { return RSVP.Promise.resolve(null); });
-  let options = { id: '1', query: {sort: '-date'} };
+  let options = {id: '1', query: {sort: '-date'}};
   let promise = adapter.find(options);
+
   assert.ok(typeof promise.then === 'function', 'returns a thenable');
-  assert.ok(adapter.findOne.calledOnce, 'findOne called');
-  assert.ok(adapter.findOne.calledWith(options.id, options.query), 'findOne called with `"1"` and query `{"sort": "-date"}`');
+  promise.then(() => {
+    assert.ok(adapter.findOne.calledOnce, 'findOne called');
+    assert.ok(
+      adapter.findOne.calledWith(options.id, options.query),
+      'findOne called with `"1"` and query `{"sort": "-date"}`'
+    );
+    done();
+  });
 });
 
+// Yeah... findQuery is used when options arg is not a string or an object.
+// undefined is not a string or an object, but so is a number.
+// TODO: id.toString() in to public methods? Or make this test more specific?
```

Comment:
One other thing, when branching off to `findOne()` we no longer care about the`id` being a string (https://github.com/pixelhandler/ember-jsonapi-resources/blob/master/addon/adapters/application.js#L52) we simply pass whatever `options.id` was, which will work since it's just concatenating. Confusingly inconsistent:

`store.find('post', 1)` -> won't work.
`store.find('post', {id: 1}) -> will work.